### PR TITLE
Fix bug with weights if unc

### PR DESCRIPTION
**Pull request by javerbukh**
If `weights` is set to "unc", it will go through this code block and - if `uncerts` is None - `weights` will be used as a list. By setting `weights` to None here, that later code block (line ~492) is not executed and the traceback is avoided.

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
I am not involved in this code, so I don't think I am qualified to review.

If this is fixing a bug, please add a regression test. Thanks!

**Comment by pllim** [comment]
And maybe a change log? 😸 

https://github.com/astropy/specutils/blob/main/CHANGES.rst

File: CHANGES.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ New Features
 Bug Fixes
 ^^^^^^^^^
 
+- Fix bug in fitting with weights if 'unc'. [#979]
```
Comment (rosteen) [inline CHANGES.rst:10]:
```suggestion
- Fix bug in fitting with weights if weights argument is set to 'unc'. [#979]
```

File: specutils/fitting/fitmodels.py [hunk 2]
```diff
@@ -382,12 +384,14 @@ def _fit_lines(spectrum, model, fitter=fitting.LevMarLSQFitter(calc_uncertaintie
...
+            if spectrum.uncertainty is not None:
+                # Convert uncertainty to StdDev, then invert
+                uncerts = StdDevUncertainty(uncertainty._convert_uncertainty(spectrum.uncertainty,
+                                                                             StdDevUncertainty))
+                # Astropy fitters expect weights in 1/sigma
                 weights = uncerts.array ** -1
             else:
+                weights = None
                 warnings.warn("Uncertainty values are not defined, but are "
                               "trying to be used in model fitting.")
```
Comment (rosteen) [inline specutils/fitting/fitmodels.py:396]: Not something you changed, but this warning could be more informative. Maybe something like "Fitting is set to use uncertainties as weights, but the input spectrum's uncertainty is None"

**Comment by javerbukh** [comment]
@rosteen Good suggestions, updated both!

**Comment by rosteen** [comment]
Thanks! Looks like one of the tests was expected that exact wording in the warning though, needs to be updated.

File: specutils/tests/test_fitting.py [hunk 3]
```diff
@@ -233,10 +236,15 @@ def calculate_rms(x, init_mod, implicit_weights):
...
         return np.median(rms)
 
-    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=True),
+    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=True, valid_uncertainty=True),
                        5.101611033799086)
-    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=False),
+    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=False, valid_uncertainty=True),
                        5.113697654869089)
+    with pytest.warns(UserWarning, match='Uncertainty values are not defined, '
+                                         'but are trying to be used in model fitting.'):
```
Comment (rosteen) [inline specutils/tests/test_fitting.py:244]:
```suggestion
    with pytest.warns(UserWarning, match='Fitting is set to use uncertainties as weights,'
                                         ' but the input spectrum's uncertainty is None'):
```

File: specutils/tests/test_fitting.py [hunk 4]
```diff
@@ -233,10 +236,15 @@ def calculate_rms(x, init_mod, implicit_weights):
...
         return np.median(rms)
 
-    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=True),
+    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=True, valid_uncertainty=True),
                        5.101611033799086)
-    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=False),
+    assert np.allclose(calculate_rms(x, init_mod, implicit_weights=False, valid_uncertainty=True),
                        5.113697654869089)
+    with pytest.warns(UserWarning, match='Fitting is set to use uncertainties as weights,'
+                                         ' but the input spectrum's uncertainty is None'):
```
Comment (rosteen) [inline specutils/tests/test_fitting.py:244]:
```suggestion
    with pytest.warns(UserWarning, match="Fitting is set to use uncertainties as weights,"
                                         " but the input spectrum's uncertainty is None"):
```

**Review by rosteen — APPROVED** [review]
Remote data failure is unrelated, approved. Thanks for the fix!

**Pull request merged**